### ISSUE TITLE
Getting a new instance from a compiled container

### ISSUE THREAD

**Issue by TRPB** [ray-di/Ray.Di#77]
I'm trying to do this:

``` php
$injector = \Ray\Di\DiCompiler::create(function() { return new Module; }, $cache, 'ray', './tmp');

$a = $injector->getInstance('A');
$a1 = $injector->getInstance('A');

var_dump($a === $a1);
```

Unfortunately the behaviour is different to using the injector alone:

``` php
$injector = Ray\Di\Injector::create([new Module]);

$a = $injector->getInstance('A');
$a1 = $injector->getInstance('A');

var_dump($a === $a1);
```

using DiCompiler returns true, using Injector returns false.  Is it possible to fetch a new instance of an object from a compiled container?

**Comment by koriym** [comment]
@TomBZombie Thanks for the reporting. I confirm the issue. It should be possible to fetch new instance of an object from a `Di Compiler`.

**Comment by koriym** [comment]
fixed with `1.4.3`
